src/services/inlayHints.ts

```diff
@@ -437,24 +438,26 @@ export function provideInlayHints(context: InlayHintsContext): InlayHint[] {
             return;
         }
 
-        for (let i = 0; i < node.parameters.length && i < signature.parameters.length; ++i) {
-            const param = node.parameters[i];
-            if (!isHintableDeclaration(param)) {
-                continue;
+        let pos = 0;
+        for (const param of node.parameters) {
+            if (isHintableDeclaration(param)) {
+                addParameterTypeHint(param, parameterIsThisKeyword(param) ? signature.thisParameter : signature.parameters[pos]);
+                if (parameterIsThisKeyword(param)) {
+                    continue;
+                }
             }
+            pos++;

```

Comment:
```suggestion    
            }
            if (parameterIsThisKeyword(param)) {
                continue;
            }
            pos++;
```

Something like this feels better to me; saves me having to think about whether or not `isHintableDeclaration` will end up false and then we get off by one errors.

Reply:
Oke, I've changed. Thanks